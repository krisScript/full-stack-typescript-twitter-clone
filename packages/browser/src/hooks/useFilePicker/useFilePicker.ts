import React, { useState, SyntheticEvent } from 'react';
interface FileData {
  fileUrl: string;
  file: File | undefined;
}
const useFilePicker = (): {
  fileData: FileData | undefined;
  fileHandler: (e: SyntheticEvent<HTMLInputElement>) => FileData;
} => {
  const [fileData, setFileData] = useState<FileData | undefined>();
  const fileHandler = (e: SyntheticEvent<HTMLInputElement>): FileData => {
    const target = e.target as HTMLInputElement;
    const file: File = (target.files as FileList)[0];
    const fileUrl = window.URL.createObjectURL(file);
    const fileData: FileData = {
      file,
      fileUrl,
    };
    setFileData(fileData);
    return fileData;
  };
  return { fileData, fileHandler };
};
export default useFilePicker;
