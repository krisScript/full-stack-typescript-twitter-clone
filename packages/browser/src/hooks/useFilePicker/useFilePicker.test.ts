import React, { useRef, SyntheticEvent } from 'react';
import { renderHook, act } from '@testing-library/react-hooks';
import useFilePicker from './useFilePicker';

window.URL.createObjectURL = jest.fn(() => 'mockString');

describe('usePortal', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it('returns a file', () => {
    const fileMock = {
      name: 'test.jpg',
      type: 'image/jpg',
    } as File;

    const fileList: FileList = {
      length: 1,
      item: () => null,
      0: fileMock,
      [Symbol.iterator]: function*() {},
    };

    const event = {
      target: {
        files: fileList,
      },
    } as React.ChangeEvent<HTMLInputElement>;
    const { result } = renderHook(() => useFilePicker());
    act(() => {
      const { fileHandler } = result.current;
      const fileData = fileHandler(event);
      expect(fileData).toMatchSnapshot();
    });
  });
});
