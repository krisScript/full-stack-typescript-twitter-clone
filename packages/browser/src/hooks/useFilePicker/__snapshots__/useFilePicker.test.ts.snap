// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePortal returns a file 1`] = `
Object {
  "file": Object {
    "name": "test.jpg",
    "type": "image/jpg",
  },
  "fileUrl": "mockString",
}
`;
